R/widgets.R

```diff
@@ -68,23 +68,39 @@ widget_diff_ <- function(before, after, width = NULL, height = NULL) {
   )
 }
 
+
+diff_text_ <- function(before, after, input) {
+  # https://github.com/brodieG/diffobj/issues/125#issuecomment-414699100
+  shiny::HTML(
+    as.character(
+      diffobj::diffChr(
+        before, 
+        after, 
+        mode = input$mode %||% "unified",
+        format = 'html',
+        style = list(html.output = 'diff.w.style')
+      )
+    )
+  )
+}
+
 #' @rdname htmlwidgets
 #' @export
-widget_toggle <- function(before, after, width = NULL, height = NULL) {
+widget_toggle <- function(before, after, input, width = NULL, height = NULL) {
```

Comment:
This new argument needs a doc bullet. Should it have a default so it doesn't look like a compulsory argument?

Reply:
Yea, `input = NULL` would be better